custom_components/tuya_local/devices/HCT-626_dual_water_timer.yaml

```diff
@@ -0,0 +1,220 @@
+name: Dual water timer
+products:
+  - id: smd9kj1n
+primary_entity:
+  entity: valve
+  name: Valve 1
+  class: water
+  dps:
+    - id: 1
+      type: boolean
+      name: valve
+secondary_entities:
+  - entity: valve
+    name: Valve 2
+    class: water
+    dps:
+      - id: 2
+        type: boolean
+        name: valve
+  - entity: number
+    name: Timer 1
+    category: config
+    class: duration
+    mode: box
+    dps:
+      - id: 17
+        type: integer
+        name: value
+        range:
+          min: 60
+          max: 3600
+        mapping:
+          - scale: 60
+            step: 60
+        unit: min
+  - entity: number
+    name: Timer 2
+    category: config
+    class: duration
+    mode: box
+    dps:
+      - id: 18
+        type: integer
+        name: value
+        range:
+          min: 60
+          max: 3600
+        mapping:
+          - scale: 60
+            step: 60
+        unit: min
+  - entity: sensor
+    name: Weather delay remaining
+    category: diagnostic
+    dps:
+      - id: 42
+        type: integer
+        name: sensor
+        unit: hour
+        readonly: true
+  - entity: switch
+    name: Smart weather delay
+    category: config
+    dps:
+      - id: 43
+        type: boolean
+        name: switch
+  - entity: sensor
+    name: Weather
+    class: enum
+    category: diagnostic
+    dps:
+      - id: 44
+        type: string
+        name: sensor
+        optional: true
+        mapping:
+          - dps_val: sunny
+            value: Sunny
+            icon: "mdi:weather-sunny"
+          - dps_val: cloudy
+            value: Cloudy
+            icon: "mdi:weather-cloudy"
+          - dps_val: rainy
+            value: Rainy
+            icon: "mdi:weather-rainy"
+          - dps_val: snowy
+            value: Snowy
+            icon: "mdi:weather-snowy"
+          - dps_val: null
+            value: Unavailable
+            icon: "mdi:weather-sunny-off"
+  - entity: select
+    name: Weather delay
+    icon: "mdi:weather-cloudy-clock"
+    category: config
+    dps:
+      - id: 45
+        type: string
+        name: option
+        mapping:
+          - dps_val: "cancel"
+            value: "Off"
+          - dps_val: "1"
+            value: "1 day"
+          - dps_val: "2"
+            value: "2 days"
+          - dps_val: "3"
+            value: "3 days"
+          - dps_val: "4"
+            value: "4 days"
+          - dps_val: "5"
+            value: "5 days"
+          - dps_val: "6"
+            value: "6 days"
+          - dps_val: "7"
+            value: "7 days"
+  - entity: sensor
+    category: diagnostic
+    class: battery
+    dps:
+      - id: 47
+        type: integer
+        name: sensor
+        mapping:
+          - scale: 0.05
+        unit: '%'
+        precision: 0
+  - entity: binary_sensor
+    category: diagnostic
+    class: battery
+    dps:
+      - id: 53
+        type: integer
```

Comment:
Binary sensor should have boolean values.  Is there a mapping needed? Is 53 a bitfield with only one bit indicating low battery, or is low battery the only indication in here (the mapping would be different for these two cases - either specifying the bit value for true, and defaulting to false, or specifying  0=false and defaulting to true).

Reply:
@make-all The DPS is an integer and I've only ever seen it be 0. But when I had it writable setting it to any value other than 0 triggered the low battery warning in the smart life app. So a mapping to a bool is probably correct.